src/main/java/bob/BobStorage.java

```diff
@@ -0,0 +1,162 @@
+package bob;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+
+/**
+ * Handles the saving of user tasks.
+ */
+public class BobStorage {
+
+    private static final String HOME_BASE_PATH = System.getProperty("user.home");
+    private static final File SAVE_DATA = new File(HOME_BASE_PATH + "/save.txt");
+    private static final String NEW_LINE = System.lineSeparator();
+
+    private BobUI ui;
+
+    public BobStorage(BobUI ui) {
+        this.ui = ui;
+    }
+
+    /**
+     * Add items to storage.
+     *
+     * @param t A Task object.
+     * @param list An ArrayList that contains the tasks.
+     * @return a task object that can be further modified.
+     */
+    public Task addItem(Task t, ArrayList<Task> list) throws BobException {
+        list.add(t);
+        this.updateTaskList(list);
+        return t;
+    }
+
+    /**
+     * Updates task list and saves them to a file.
+     *
+     * @param list An ArrayList that contains the tasks.
+     */
+    public void updateTaskList(ArrayList<Task> list) {
+        try {
+            if (!this.SAVE_DATA.exists()) {
+                this.instantiateDirectory();
+            }
+            FileWriter fileWriter = new FileWriter(this.SAVE_DATA, false);
+            for (Task t : list) {
+                fileWriter.write(t.toSavableFormat() + this.NEW_LINE);
+            }
+            fileWriter.close();
+
+        } catch (Exception e) {
+            this.ui.getErrorText(
+                    new BobException.FileAccessError("An error occurred when trying to access the save file."));
+        }
+    }
+
+    /**
+     * Load tasks from saved file.
+     *
+     * @return List of tasks from the save file.
+     * @throws BobException.FileAccessError If file is not accessible.
+     * @throws BobException.CorruptedSaveData If save file contains invalid task format.
+     */
+    public ArrayList<Task> loadSavedTasks() throws BobException.FileAccessError, BobException.CorruptedSaveData {
+
+        ArrayList<Task> savedTasks = new ArrayList<>();
+
+        try {
+
+            if (!this.SAVE_DATA.exists()) {
+                this.instantiateDirectory();
+                return new ArrayList<Task>();
+            }
+
+            try (BufferedReader br = new BufferedReader(new FileReader(this.SAVE_DATA))) {
+
+                String line;
+                boolean invalidFile = false;
```

Comment:
According to our coding standard, it is recommended as to use a prefix such as _is, has, was,_ etc. for boolean variable/method names.
Maybe `isValidFile` is a better name for this variable?